### PR TITLE
Adds bundle options to the package manifest

### PR DESCRIPTION
**Pull request by Shazwazza**
to more control over how bundling works for static file processing for app_plugins.

See https://github.com/umbraco/Umbraco-CMS/discussions/10469#discussioncomment-903273 for discussion.

The package.manifest has a new option: "bundleOptions:" The values can be:

* "Default" - which is the default value if none is specified and means that the package's assets will be processed within the main package bundle like it does today. (in debug, files will not be processed)
* "None" - which means that the assets in the package will not be processed at all and will all be requested as individual assets. This will essentially be a bundle that has composite processing turned off for both debug and production.
* "Independent" - which means the packages assets will be processed as it's own separate bundle. (in debug, files will not be processed)

## Other changes

* Added BundlingOptions - so that we can have more control over bundling behavior
* Simplifies the SmidgeRuntimeBundler with how it creates bundle options and base them on BundlingOptions
* Fixes CSS bundling to ensure that in debug mode we are using composite files

## Testing

* Create a couple App_Plugins with package.manifests declaring both CSS and JS (don't add the bundleOptions parameter yet). 
* Run umbraco and ensure those JS/CSS are loaded in both Debug and not Debug modes. In debug mode they will be their normal paths, in non-debug they will be bundled.
* Change the package manifest to include "bundleOptions": "None"
* Run umbraco and ensure those JS/CSS are loaded in both Debug and not Debug modes. In both modes, the assets will be loaded as their original paths but they will also include the cache buster query string
* Change the package manifest to include "bundleOptions": "Independent"
* Run umbraco and ensure those JS/CSS are loaded in both Debug and not Debug modes. In debug mode they will be their normal paths, in non-debug they will be bundled, but there will be a separate bundle for each package.

### CODE REVIEW

File: src/Umbraco.Core/Manifest/IManifestParser.cs [hunk 1]
```diff
@@ -4,13 +4,13 @@ namespace Umbraco.Cms.Core.Manifest
 {
     public interface IManifestParser
     {
-        string Path { get; set; }
+        //string Path { get; set; }
```
[the remark this answers is not in the document]
Comment (Shazwazza) [reply src/Umbraco.Core/Manifest/IManifestParser.cs:7]: Doh, I meant to undo this, i'll fix up.

**Comment by Shazwazza** [comment]
@nikolajlauridsen thanks! good catches on those, I've pushed updates.

**Comment by nikolajlauridsen** [comment]
When testing this I found something I think is a bug, the "Default" and "None" options tested out fine, however when I got to "Independent" I got an error from trying to bundle the CSS:

```
System.InvalidOperationException: /App_Plugins/CustomWelcomeDashboard/customwelcomedashboard.css(1,1-2): run-time error JS1195: Expected expression: .,/App_Plugins/CustomWelcomeDashboard/customwelcomedashboard.css(1,19): run-time
 error JS1004: Expected ';',/App_Plugins/CustomWelcomeDashboard/customwelcomedashboard.css(1,22):{continues...}
``` 

For testing I just followed the custom dashboard tutorial, so nothing super fancy, I get the error on both dashboard packages, the CSS is nothing special and should be valid:

```CSS
.welcome-dashboard h1 {
  font-size:4em;
  color:gold;
}

.welcome-dashboard p{
  color: gold;
}

.welcome-dashboard{
  background-color: navy;
}
```

This of course caused the CSS bundles to be empty so no styles.

**Comment by nathanwoulfe** [comment]
Wouldn't be due to the missing space before the opening brace in the second and third rules? Pretty sure that's still valid CSS, but might be causing issues...

**Comment by nikolajlauridsen** [comment]
> Wouldn't be due to the missing space before the opening brace in the second and third rules? Pretty sure that's still valid CSS, but might be causing issues...

Hmm, I tried adding spaces, but it still throws the same exception

**Comment by Shazwazza** [comment]
The error indicates that a JS engine is trying to parse CSS.

It's because in `RegisterPackageBundlesForIndependentOptions` we are accidentally calling `CreateJsBundle` even for css asset types. I'll fix up.

**Comment by Shazwazza** [comment]
Just pushed a fix.

**Review by nikolajlauridsen — APPROVED** [review]
Looks all good now, and tests out good as well 👍 I'll go ahead and merge 🎉

**Pull request merged**